bmxlib-cbmx/src/writer.cpp

```diff
@@ -188,6 +192,30 @@ void bmx_channel_count(void* bmx_writer, int track_index, int channel_count)
     writer->tracks[track_index]->SetChannelCount(channel_count);
 }
 
+void bmx_avci_header(void* bmx_writer, int track_index, int ps_avcihead, EssenceType essence_type)
+{
+    unsigned char avci_header_data[AVCI_HEADER_SIZE];
+    BmxWriter* writer = (BmxWriter*)bmx_writer;
+    bmx::Rational frame_rate = writer->clip->GetFrameRate();
+    switch(essence_type) {
+        case AVCI200_1080I:
+        case AVCI200_1080P:
+        case AVCI200_720P:
+        case AVCI100_1080I:
+        case AVCI100_1080P:
+        case AVCI100_720P:
+        case AVCI50_1080I:
+        case AVCI50_1080P:
+        case AVCI50_720P:
+            if (ps_avcihead && bmx::get_ps_avci_header_data((bmx::EssenceType)essence_type, frame_rate,
+                                                    avci_header_data, sizeof(avci_header_data)))
+            {
+                writer->tracks[track_index]->SetAVCIHeader(avci_header_data, sizeof(avci_header_data));
+            }
+            break;
```

Comment:
add a default value and return false.
it's better to return a boolean to know if it's applyed correctly.